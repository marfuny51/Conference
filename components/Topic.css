.Table {
    text-align: center;
}

.TableTopic {
    border-collapse: collapse;
    width: 70vw;
    height: 10vh;
    margin: 6vh auto 2vh auto;
    table-layout: fixed;
}

.TableTopicAdd {
    border-collapse: collapse;
    width: 70vw;
    height: 10vh;
    margin: 6vh auto 2vh auto;
    opacity:0.3;
    table-layout: fixed;
}

.TableHead {
    color: #00D970;
}

.TableHead th {
    font-family: 'Muli', sans-serif;
    font-size: 1.2vw;
    background-color: #127a16;
    color: white;
    letter-spacing: 0.1vw;
    padding: 1vw;
    border: 0vw;
}

.TableBody tr {
    height: 5vw;
    background-color: #d3f1d4;
    border-bottom: 0.15vw solid #127a16;
}

.TableBody tr:hover {
    background-color: #f5f5f5;
}

.TableBody tr td {
    width: 13vw;
    font-family: 'Muli', sans-serif;
    font-size: 1.1vw;
    border: 0vw;
    padding: 0.5vw;
}

.Read {
    font-family: 'Muli', sans-serif;
    font-size: 1vw;
    height: 2vw;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
}

.Read:hover {
    font-weight: bold;
    border:0.2vw solid #456835;
    cursor: pointer;
    transform: scale(1.03);
}

.ReadNone {
    display: none;
}

.Read2 {
    font-family: 'Muli', sans-serif;
    font-size: 1vw;
    height: 2vw;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
    margin-right: 1.2vw;
}

.Read2:hover {
    border:0.2vw solid #456835;
    cursor: pointer;
    transform: scale(1.03);
}

.Add {
    font-family: 'Muli', sans-serif;
    font-size: 1.1vw;
    font-weight: bold;
    height: 3vw;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
}

.Add:hover {
    transform: scale(1.03);
    border:0.2vw solid #456835;
    cursor: pointer;
}

.NavLink {
    color: #172c11;
    font-weight: bold;
}

.LoadData {
    font-size: 1.2vw;
    text-align: center;
    color: white;
}

.AddTopic {
    width: 48vw;
    height: 24vw;
    position: absolute;
    top: 35%;
    left: 26%;
    background-color: #d3f1d4;
    border: 0.2vw solid #77b45b;
    border-radius: 0.5vw;
}

.Title {
    font-family: 'Muli', sans-serif;
    font-weight: bold;
    letter-spacing:0.1vw;
    padding-top: 1vw;
}

.Input {
    font-family: 'Muli', sans-serif;
    font-size: 1.1vw;
    text-align: left;
    padding: 3.8vh 1vw 3vh 4vw;
}

.Input input{
    font-size: 1vw;
    width: 9vw;
    height: 1.2vw;
}

.Input textarea{
    font-size: 1vw;
    width: 16vw;
    height: 10vw;
}

.Valide {
    color: red;
}

.Textarea {
    width: 18vw;
    height: 18vh;
}

.Hide {
    width: 6.2vw;
    font-family: 'Muli', sans-serif;
    font-size: 1vw;
    background-color: #2f4f30;
    color: white;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
    margin-bottom: 1vh;
}

.Hide:hover {
    border:0.2vw solid #d3f1d4;
    cursor: pointer;
    transform: scale(1.03);
}

.ReadDiv {
    width: 27vw;
    height: 0vw;
    margin: auto;
    opacity: 0;
    display: none;
}

.ReadDivAnim1 {
    display: block;
    transition-duration: 0.7s;
    -webkit-transition-timing-function: linear; 
    transition-timing-function: linear;
    overflow: hidden;
    width: 27vw;
    height: 10vw; 
    opacity: 1;  
}

.ReadDivAnim2 {
    transition-duration: 0.7s;
    -webkit-transition-timing-function: linear; 
    transition-timing-function: linear;
    overflow: hidden;
    width: 27vw;
    height: 0vw; 
    opacity: 0;  
}

.TableBody .LinkTd {
    font-weight: bold;
}

.col1 {
    width: 15vw;
}

.col2 {
    width: 10vw;
}

.col3 {
    width: 27vw;
}

.col4 {
    width: 6vw;
}