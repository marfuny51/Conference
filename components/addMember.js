import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from './events';

import './Member.css';

class AddMember extends React.PureComponent {

  static propTypes = {
    member:PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        phone: PropTypes.string.isRequired,
        email: PropTypes.string.isRequired,
        job: PropTypes.string.isRequired,
    }),
  };

  state = {
    newName: '',
    newPhone: '',
    newEmail: '',
    newJob: '',
    errorName:'',
    errorPhone:'',
    errorEmail: '',
    errorJob:'',
    valideName: true,
    validePhone: true,
    valideEmail: true,
    valideJob: true,
  }

  componentDidMount() {
    if (!this.state.newName.match(/^[a-zA-Z'][a-zA-Z-' ]+[a-zA-Z']?$/)) {
        this.setState({errorName:'Name should includes only letters!', valideName: false}); 
    }
    else {
        this.setState({errorName:'', valideName: true});
    } 
    if (!this.state.newPhone.match(/(\+375|80)(29|25|44|33)(\d{3})(\d{2})(\d{2})$/)) {
        this.setState({errorPhone:'Phone should be the same form: +375291112233', validePhone: false}); 
    }
    else {
        this.setState({errorPhone:'', validePhone: true});
    } 
    if (!this.state.newEmail.match(/^[\w]{1}[\w-\.]*@[\w-]+\.[a-z]{2,4}$/)) {
      this.setState({errorEmail:'Email should be like hor@mail.com!', valideEmail: false}); 
    }
    else {
      this.setState({errorEmail:'', valideEmail: true});
    }
    if (!this.state.newJob.match(/^.{2,}/)) {
        this.setState({errorJob:'Your job should includes not less then 2 letters!', valideJob: false}); 
    }
    else {
        this.setState({errorJob:'', valideJob: true});
    } 
  }

  setNewName = (EO) => {
    this.setState({newName: EO.target.value}, this.error);
  }

  setNewPhone = (EO) => {
    this.setState({newPhone: EO.target.value}, this.error);
  }

  setNewEmail = (EO) => {
    this.setState({newEmail: EO.target.value}, this.error);
  }

  setNewJob = (EO) => {
    this.setState({newJob: EO.target.value}, this.error);
  }

  error = () => {
    if (!this.state.newName.match(/^[a-zA-Z'][a-zA-Z-' ]+[a-zA-Z']?$/)) {
        this.setState({errorName:'Name should includes only letters!', valideName: false}); 
    }
    else {
        this.setState({errorName:'', valideName: true});
    }
    if (!this.state.newPhone.match(/(\+375|80)(29|25|44|33)(\d{3})(\d{2})(\d{2})$/)) {
      this.setState({errorPhone:'Phone should be the same form: +375291112233', validePhone: false}); 
    }
    else {
      this.setState({errorPhone:'', validePhone: true});
    } 
    if (!this.state.newEmail.match(/^[\w]{1}[\w-\.]*@[\w-]+\.[a-z]{2,4}$/)) {
      this.setState({errorEmail:'Email should be like hor@mail.com!', valideEmail: false}); 
    }
    else {
      this.setState({errorEmail:'', valideEmail: true});
    } 
    if (!this.state.newJob.match(/^.{2,}/)) {
      this.setState({errorJob:'Your job should includes not less then 2 letters!', valideJob: false}); 
    }
    else {
      this.setState({errorJob:'', valideJob: true});
    } 
  }

  save = () => {
      let name=this.state.newName;
      let phone=this.state.newPhone;
      let email=this.state.newEmail;
      let job=this.state.newJob;
      voteEvents.emit('ESaveMember', this.props.id, name, phone, email, job);
  }

  cancel = () => {
    voteEvents.emit('ECancelMember');
  }

  render() {
    
    return (
        <div key = {this.props.id} className = 'AddMember'>
            <div className='Title'>Please, enter info</div>
            <div className='Input'>
              <span>Name: </span><input type='text' defaultValue={this.props.name} onChange={this.setNewName}/><span className='Valide'>{this.state.errorName}</span><br/>
              <span>Phone: </span><input type='text'defaultValue={this.props.phone} onChange={this.setNewPhone}/><span className='Valide'>{this.state.errorPhone}</span><br/>
              <span>Email: </span><input type='text'defaultValue={this.props.email} onChange={this.setNewEmail}/><span className='Valide'>{this.state.errorEmail}</span><br/>
              <span>Job: </span><input type='text'defaultValue={this.props.job} onChange={this.setNewJob}/><span className='Valide'>{this.state.errorJob}</span><br/>
            </div>
            <input type="button" value="Save" className='Read2' onClick={this.save} disabled = {(this.state.valideName&&this.state.validePhone&&this.state.valideEmail&&this.state.valideJob)?false:true}/>
            <input type="button" value="Cancel" className='Read2' onClick={this.cancel}/>
        </div>
    );

  }

}

export default AddMember;
