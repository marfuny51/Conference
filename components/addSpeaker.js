import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from './events';

//import './editadd.css';

class AddSpeaker extends React.PureComponent {

  static propTypes = {
    speaker:PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        phone: PropTypes.string.isRequired,
        position: PropTypes.string.isRequired,
        topic: PropTypes.string.isRequired,
    }),
    topic:PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      mainWords: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      info: PropTypes.string.isRequired,
    }),
  };

  state = {
    newName: '',
    newPhone: '',
    newPosition: '',
    newTopic: '',
    newTitle: '',
    newMainWords: '',
    newInfo: '',
    errorName: '',
    errorPhone: '',
    errorPosition: '',
    errorTopic: '',
    errorTitle:'',
    errorMainWords:'',
    errorInfo:'',
    valideName: true,
    validePhone: true,
    validePosition: true,
    valideTopic: true,
    valideTitle: true,
    valideMainWords: true,
    valideInfo: true,
  }

  componentDidMount() {
    if (!this.state.newName.match(/^[A-Za-z]+$/)) {
        this.setState({errorName:'Name should includes only letters!', valideName: false}); 
    }
    else {
        this.setState({errorName:'', valideName: true});
    }
    if (!this.state.newPhone.match(/^(\s*)?(\+)?([- _():=+]?\d[- _():=+]?){10,14}(\s*)?$/)) {
        this.setState({errorPhone:'Phone should be the same form: +375(**)***-**-**', validePhone: false}); 
    }
    else {
        this.setState({errorPhone:'', validePhone: true});
    }
    if (!this.state.newPosition.match(/^[A-Za-z]+$/)) {
        this.setState({errorPosition:'Your position should includes only letters!', validePosition: false}); 
    }
    else {
        this.setState({errorPosition:'', validePosition: true});
    }
    if (!this.state.newTopic.match(/^[A-Za-z]+$/)) {
        this.setState({errorTopic:'Name of your topic should includes only letters!', valideTopic: false}); 
    }
    else {
        this.setState({errorTopic:'', valideTopic: true});
    } 
    if (!this.state.newTitle.match(/^[a-fA-F0â€“9]{3,}/)) {
      this.setState({errorTitle:'Title should includes more then 3 letters!', valideTitle: false}); 
    }
    else {
      this.setState({errorTitle:'', valideTitle: true});
    }
    if (!this.state.newMainWords.match(/^[A-Za-z0-9]{20,}/)) {
      this.setState({errorMainWords:'Key words should includes more then 20 letters!', valideMainWords: false}); 
    }
    else {
      this.setState({errorMainWords:'', valideMainWords: true});
    } 
    if (!this.state.newInfo.match(/^[A-Za-z0-9]{30,}/)) {
      this.setState({errorInfo:'Info should includes more then 30 letters!', valideInfo: false}); 
    }
    else {
      this.setState({errorInfo:'', valideInfo: true});
    } 
}

  setNewName = (EO) => {
    this.setState({newName: EO.target.value}, this.error);
  }

  setNewPhone = (EO) => {
    this.setState({newPhone: EO.target.value}, this.error);
  }

  setNewPosition = (EO) => {
    this.setState({newPosition: EO.target.value}, this.error);
  }

  setNewTopic = (EO) => {
    this.setState({newTopic: EO.target.value}, this.error);
  }

  setNewTitle = (EO) => {
    this.setState({newTitle: EO.target.value}, this.error);
  }

  setNewMainWords = (EO) => {
    this.setState({newMainWords: EO.target.value}, this.error);
  }

  setNewInfo = (EO) => {
    this.setState({newInfo: EO.target.value}, this.error);
  }

  error = () => {
    if (!this.state.newName.match(/^[A-Za-z]+$/)) {
        this.setState({errorName:'Name should includes only letters!', valideName: false}); 
    }
    else {
        this.setState({errorName:'', valideName: true});
    }
    if (!this.state.newPhone.match(/^(\s*)?(\+)?([- _():=+]?\d[- _():=+]?){10,14}(\s*)?$/)) {
        this.setState({errorPhone:'Phone should be the same form: +375(**)***-**-**', validePhone: false}); 
    }
    else {
        this.setState({errorPhone:'', validePhone: true});
    }
    if (!this.state.newPosition.match(/^[A-Za-z]+$/)) {
        this.setState({errorPosition:'Your position should includes only letters!', validePosition: false}); 
    }
    else {
        this.setState({errorPosition:'', validePosition: true});
    }
    if (!this.state.newTopic.match(/^[A-Za-z]+$/)) {
        this.setState({errorTopic:'Name of your topic should includes only letters!', valideTopic: false}); 
    }
    else {
        this.setState({errorTopic:'', valideTopic: true});
    }
    if (!this.state.newTitle.match(/^[A-Za-z0-9]{3,}/)) {
      this.setState({errorTitle:'Title should includes more then 3 letters!', valideTitle: false}); 
    }
    else {
      this.setState({errorTitle:'', valideTitle: true});
    } 
    if (!this.state.newMainWords.match(/^[A-Za-z0-9]{20,}/)) {
      this.setState({errorMainWords:'Key words should includes more then 20 letters!', valideMainWords: false}); 
    }
    else {
      this.setState({errorMainWords:'', valideMainWords: true});
    } 
    if (!this.state.newInfo.match(/^[A-Za-z0-9]{30,}/)) {
      this.setState({errorInfo:'Info should includes more then 30 letters!', valideInfo: false}); 
    }
    else {
      this.setState({errorInfo:'', valideInfo: true});
    } 
  }

  save = () => {
    if (this.props.mode ===2) {
      let name=this.state.newName;
      let phone=this.state.newPhone;
      let position=this.state.newPosition;
      let topic=this.state.newTopic;
      voteEvents.emit('ESave', this.props.id, name, phone, position, topic);
    }
    if (this.props.mode ===4) {
      let title=this.state.newTitle;
      let mainWords=this.state.newMainWords;
      let author=this.state.newName;
      let info=this.state.newInfo;
      voteEvents.emit('ESave', this.props.id, title, mainWords, author, info);
    }
  }

  cancel = () => {
    voteEvents.emit('ECancel');
  }

  render() {
    
    return (
        
        <div>
          {
          (this.props.mode===2)&&
          <div key = {this.props.id}>
            <span>Please, enter info</span><br/>
            <span>Name: </span><input type='text' defaultValue={this.props.name} onChange={this.setNewName}/><span>{this.state.errorName}</span><br/>
            <span>Phone: </span><input type='text'defaultValue={this.props.phone} onChange={this.setNewPhone}/><span>{this.state.errorPhone}</span><br/>
            <span>Position: </span><input type='text'defaultValue={this.props.position} onChange={this.setNewPosition}/><span>{this.state.errorPosition}</span><br/>
            <span>Topic: </span><input type='text'defaultValue={this.props.topic} onChange={this.setNewTopic}/><span>{this.state.errorTopic}</span><br/>
            <input type="button" value="Save" onClick={this.save} disabled = {(this.state.valideName&&this.state.validePhone&&this.state.validePosition&&this.state.valideTopic)?false:true}/>
            <input type="button" value="Cancel" onClick={this.cancel}/>
            </div>
          }
           {
          (this.props.mode===4)&&
          <div key = {this.props.id}>
            <span>Please, enter info</span><br/>
            <span>Title: </span><input type='text' defaultValue={this.props.title} onChange={this.setNewTitle}/><span>{this.state.errorTitle}</span><br/>
            <span>Key words: </span><input type='text'defaultValue={this.props.mainWords} onChange={this.setNewMainWords}/><span>{this.state.errorMainWords}</span><br/>
            <span>Author: </span><input type='text'defaultValue={this.props.author} onChange={this.setNewName}/><span>{this.state.errorName}</span><br/>
            <span>Info: </span><textarea defaultValue={this.props.info} onChange={this.setNewInfo}/><span>{this.state.errorInfo}</span><br/>
            <input type="button" value="Save" onClick={this.save} disabled = {(this.state.valideTitle&&this.state.valideName&&this.state.valideMainWords&&this.state.valideInfo)?false:true}/>
            <input type="button" value="Cancel" onClick={this.cancel}/>
          </div>
      }

         
        
        </div>

        
 
    );

  }

}

export default AddSpeaker;
