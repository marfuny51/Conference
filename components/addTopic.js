import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from './events';

//import './editadd.css';

class AddTopic extends React.PureComponent {

  static propTypes = {
    topic:PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      mainWords: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      info: PropTypes.string.isRequired,
    }),
  };

  state = {
    newName: '',
    newTitle: '',
    newMainWords: '',
    newInfo: '',
    errorName: '',
    errorTitle:'',
    errorMainWords:'',
    errorInfo:'',
    valideName: true,
    valideTitle: true,
    valideMainWords: true,
    valideInfo: true,
  }

  componentDidMount() {
    if (!this.state.newName.match(/^[A-Za-z]+$/)) {
        this.setState({errorName:'Name should includes only letters!', valideName: false}); 
    }
    else {
        this.setState({errorName:'', valideName: true});
    }
    if (!this.state.newTitle.match(/^[a-fA-F0â€“9]{3,}/)) {
      this.setState({errorTitle:'Title should includes more then 3 letters!', valideTitle: false}); 
    }
    else {
      this.setState({errorTitle:'', valideTitle: true});
    }
    if (!this.state.newMainWords.match(/^[A-Za-z0-9]{20,}/)) {
      this.setState({errorMainWords:'Key words should includes more then 20 letters!', valideMainWords: false}); 
    }
    else {
      this.setState({errorMainWords:'', valideMainWords: true});
    } 
    if (!this.state.newInfo.match(/^[A-Za-z0-9]{30,}/)) {
      this.setState({errorInfo:'Info should includes more then 30 letters!', valideInfo: false}); 
    }
    else {
      this.setState({errorInfo:'', valideInfo: true});
    } 
}

  setNewName = (EO) => {
    this.setState({newName: EO.target.value}, this.error);
  }

  setNewTitle = (EO) => {
    this.setState({newTitle: EO.target.value}, this.error);
  }

  setNewMainWords = (EO) => {
    this.setState({newMainWords: EO.target.value}, this.error);
  }

  setNewInfo = (EO) => {
    this.setState({newInfo: EO.target.value}, this.error);
  }

  error = () => {
    if (!this.state.newName.match(/^[A-Za-z]+$/)) {
        this.setState({errorName:'Name should includes only letters!', valideName: false}); 
    }
    else {
        this.setState({errorName:'', valideName: true});
    }
    if (!this.state.newTitle.match(/^[A-Za-z0-9]{3,}/)) {
      this.setState({errorTitle:'Title should includes more then 3 letters!', valideTitle: false}); 
    }
    else {
      this.setState({errorTitle:'', valideTitle: true});
    } 
    if (!this.state.newMainWords.match(/^[A-Za-z0-9]{20,}/)) {
      this.setState({errorMainWords:'Key words should includes more then 20 letters!', valideMainWords: false}); 
    }
    else {
      this.setState({errorMainWords:'', valideMainWords: true});
    } 
    if (!this.state.newInfo.match(/^[A-Za-z0-9]{30,}/)) {
      this.setState({errorInfo:'Info should includes more then 30 letters!', valideInfo: false}); 
    }
    else {
      this.setState({errorInfo:'', valideInfo: true});
    } 
  }

  save = () => {
      let title=this.state.newTitle;
      let mainWords=this.state.newMainWords;
      let author=this.state.newName;
      let info=this.state.newInfo;
      voteEvents.emit('ESave', this.props.id, title, mainWords, author, info);
  }

  cancel = () => {
    voteEvents.emit('ECancel');
  }

  render() {
    
    return (
        <div key = {this.props.id}>
            <span>Please, enter info</span><br/>
            <span>Title: </span><input type='text' defaultValue={this.props.title} onChange={this.setNewTitle}/><span>{this.state.errorTitle}</span><br/>
            <span>Key words: </span><input type='text'defaultValue={this.props.mainWords} onChange={this.setNewMainWords}/><span>{this.state.errorMainWords}</span><br/>
            <span>Author: </span><input type='text'defaultValue={this.props.author} onChange={this.setNewName}/><span>{this.state.errorName}</span><br/>
            <span>Info: </span><textarea defaultValue={this.props.info} onChange={this.setNewInfo}/><span>{this.state.errorInfo}</span><br/>
            <input type="button" value="Save" onClick={this.save} disabled = {(this.state.valideTitle&&this.state.valideName&&this.state.valideMainWords&&this.state.valideInfo)?false:true}/>
            <input type="button" value="Cancel" onClick={this.cancel}/>
        </div>
    );

  }

}

export default AddTopic;
