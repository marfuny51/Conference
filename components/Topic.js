import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import {voteEvents} from './events';

import './Topic.css';

class Topic extends React.PureComponent {

  static propTypes = {
    topic:PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      mainWords: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      info: PropTypes.string.isRequired,
    }),
  };

  deleteTopic = () => {
    voteEvents.emit('EDeleteTopic', this.props.topic.id);
  }

  editTopic = () => {
    voteEvents.emit('EEditTopic', this.props.topic.id);
  }

  moreRead = () => {
    voteEvents.emit('EReadMore', this.props.topic.id);
  }

  render() {
    let info = <div className='ReadDiv'>{this.props.topic.info}<br/><input type="button" value="Hide" className='Read' onClick={this.hide}/></div>; 
    let readMore = <input type="button" value="Read more..." className='Read' onClick={this.moreRead}/>;
    return (
      <tr key={this.props.topic.id}>
        <td><NavLink to={"/topic/"+this.props.topic.title} className='NavLink'>{this.props.topic.title}</NavLink></td>
        <td>{this.props.topic.mainWords}</td>
        <td>{this.props.topic.author}</td>
        <td>{(this.props.mode===55)?info:readMore}</td>
        <td><input type="button" value="Edit" className='Read' disabled = {(this.props.mode===2)?true:false} onClick={this.editTopic}/></td>
        <td><input type="button" value="Delete" className='Read' disabled = {(this.props.mode===2)?true:false} onClick={this.deleteTopic}/></td>
      </tr>
    );

  }

}

export default Topic;
