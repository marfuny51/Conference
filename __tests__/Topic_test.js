import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from '../components/events';
import Topic from '../components/Topic';
import renderer from 'react-test-renderer';

test('работа MobileCompany', () => {

    let topic={id:3, title:'How to be happy?', mainWords: 'Happiness', author: 'Ivan Ivanov', info:'Do not worry, be happy'};
    // создаём тестовую версию компонента
    const component = renderer.create(
      <Topic 
      topic={topic}
    />
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента кнопку фильтрации 'Blocked'
    const buttonRead = component.root.findByProps(  {value:'Read more...'}); 
    // и "нажмём" на неё
    buttonRead.props.onClick();
  
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента кнопку фильтрации 'Blocked'
    const buttonHide = component.root.findByProps(  {value:'Hide'}); 
    // и "нажмём" на неё
    buttonHide.props.onClick();
  
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
  
    // найдём в вёрстке компонента кнопку 'Edit'
    const buttonEdit = component.root.findByProps( {value:'Edit'}); 
    // и "нажмём" на неё
    buttonEdit.props.onClick();
    window.confirm = () => {true};
 
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
  
    // найдём в вёрстке компонента кнопку 'Delete'
    const buttonDelete = component.root.findByProps( {value:'Delete'}); 
    // и "нажмём" на неё
    buttonDelete.props.onClick();
  
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    
      
    /*
    // можно эмулировать события, передавая в качестве объекта события то что нам нужно:
    wrapper.find('select').simulate('change', {
      target: { value: "hello" },
    });
    */
  
  });
  