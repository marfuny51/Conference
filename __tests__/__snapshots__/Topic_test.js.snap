// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Topic 1`] = `
<tr>
  <td
    className="LinkTd"
  >
    How to be happy?
  </td>
  <td>
    Happiness
  </td>
  <td>
    Ivan Ivanov
  </td>
  <td>
    <div
      className="ReadDiv"
    >
      Do not worry, be happy
      <br />
      <input
        className="Hide"
        onClick={[Function]}
        type="button"
        value="Hide"
      />
    </div>
    <input
      className="Read"
      onClick={[Function]}
      type="button"
      value="Read more..."
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Topic 2`] = `
<tr>
  <td
    className="LinkTd"
  >
    How to be happy?
  </td>
  <td>
    Happiness
  </td>
  <td>
    Ivan Ivanov
  </td>
  <td>
    <div
      className="ReadDivAnim1"
    >
      Do not worry, be happy
      <br />
      <input
        className="Hide"
        onClick={[Function]}
        type="button"
        value="Hide"
      />
    </div>
    <input
      className="ReadNone"
      onClick={[Function]}
      type="button"
      value="Read more..."
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Topic 3`] = `
<tr>
  <td
    className="LinkTd"
  >
    How to be happy?
  </td>
  <td>
    Happiness
  </td>
  <td>
    Ivan Ivanov
  </td>
  <td>
    <div
      className="ReadDivAnim2"
    >
      Do not worry, be happy
      <br />
      <input
        className="Hide"
        onClick={[Function]}
        type="button"
        value="Hide"
      />
    </div>
    <input
      className="Read"
      onClick={[Function]}
      type="button"
      value="Read more..."
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Topic 4`] = `
<tr>
  <td
    className="LinkTd"
  >
    How to be happy?
  </td>
  <td>
    Happiness
  </td>
  <td>
    Ivan Ivanov
  </td>
  <td>
    <div
      className="ReadDivAnim2"
    >
      Do not worry, be happy
      <br />
      <input
        className="Hide"
        onClick={[Function]}
        type="button"
        value="Hide"
      />
    </div>
    <input
      className="Read"
      onClick={[Function]}
      type="button"
      value="Read more..."
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Topic 5`] = `
<tr>
  <td
    className="LinkTd"
  >
    How to be happy?
  </td>
  <td>
    Happiness
  </td>
  <td>
    Ivan Ivanov
  </td>
  <td>
    <div
      className="ReadDivAnim2"
    >
      Do not worry, be happy
      <br />
      <input
        className="Hide"
        onClick={[Function]}
        type="button"
        value="Hide"
      />
    </div>
    <input
      className="Read"
      onClick={[Function]}
      type="button"
      value="Read more..."
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;
